openapi/types.rs

```diff
@@ -0,0 +1,66 @@
+use ordered_float::OrderedFloat;
+use serde::{Deserialize, Serialize};
+use utoipa::{IntoParams, ToSchema};
+use uuid::Uuid;
+
+use crate::structs::{AssetStatus, Location};
+
+/// Request to create an aircraft.
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, IntoParams)]
+pub struct RegisterAircraftPayload {
+    pub name: Option<String>,
+    /// The UUID of an [`AssetGroup`] struct, if available.
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub whitelist: Vec<Uuid>,
+    pub status: AssetStatus,
+    /// The aircraft's manufacturer.
+    ///
+    /// TODO: For now we can just say "Boeing", "Airbus", etc. Later, we
+    /// can a struct for this and store the manufacturer's name, logo,
+    /// etc.
+    pub manufacturer: String,
+    pub model: String,
+    pub serial_number: String,
+    /// The aircraft's registration number.
+    ///
+    /// In the US, this is the N number.
+    ///
+    /// This is a unique identifier for
+    /// the aircraft that can be used to look up information about the
+    /// aircraft from national aviation authorities like the FAA.
+    pub registration_number: String,
+    pub description: Option<String>,
+    pub max_payload_kg: OrderedFloat<f32>,
+    pub max_range_km: Option<OrderedFloat<f64>>,
+}
+
+/// Request to create a vertiport.
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, IntoParams)]
+pub struct RegisterVertiportPayload {
+    pub name: Option<String>,
+    /// The UUID of an [`AssetGroup`] struct, if available.
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub whitelist: Vec<Uuid>,
+    pub status: AssetStatus,
+    pub description: Option<String>,
+    pub location: Location,
+    pub vertipads: Vec<Uuid>,
```

Comment:
Adding the vertipads directly to the vertiport this way, makes it a chicken/ egg problem.
Vertipads currently require a vertiport_id to be provided (this link is on the side of the pad, not the port).
When creating new vertiports with vertipads, I suggest the following workflow:

1. Create a new vertiport
2. Use new vertiport id to create new vertipad(s)

This also makes it easier to add new pads to a port later, as you only need to use the `insert_vertipad` function providing the vertiport id

Reply:
[967e4dc](https://github.com/Arrow-air/svc-assets/pull/3/commits/967e4dc738e4d39aaa2dd762369ff7957bc3237d)